service-api/module/Application/src/Controller/YotiController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Controller;
+
+use Application\Fixtures\DataImportHandler;
+use Application\Yoti\YotiServiceInterface;
+use Laminas\Mvc\Controller\AbstractActionController;
+use Laminas\Http\Response;
+use Application\View\JsonModel;
+
+/**
+ * @psalm-suppress PropertyNotSetInConstructor
+ * Needed here due to false positive from Laminas’s uninitialised properties
+ */
+class YotiController extends AbstractActionController
+{
+    public function __construct(
+        private readonly YotiServiceInterface $yotiService,
+        private readonly DataImportHandler $dataImportHandler,
+    ) {
+    }
+
+    /**
+     * @return JsonModel
+     * @psalm-suppress PossiblyInvalidArgument
+     */
+    public function findPostOfficeAction(): JsonModel
+    {
+        $branches = [];
+        $data = json_decode($this->getRequest()->getContent(), true);
+
+        if (! isset($data["search_string"])) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(['error' => 'Missing postCode']);
+        }
+        $branchList = $this->yotiService->postOfficeBranch($data["search_string"]);
```

Comment:
`postOfficeBranch` throws exceptions but they aren't caught here at all, which means they'll just show up as generic 500s. We should at least log the error message so we can debug issues.